Long context

aidarmyrzakhan opened the pull request:
1. **Added [Frames-Benchmark dataset](https://huggingface.co/datasets/google/frames-benchmark)**  
   1.1 Configuration using the KILT  
   1.2 Configuration using the Wikipedia dump 2023  
   1.3 Configuration using the Oracle text from Frames Wikipedia links  

2. **Added [NarrativeQA dataset](https://huggingface.co/datasets/deepmind/narrativeqa)**  

3. **Added `use_middle_truncation` function** in `models/generators/generator.py` to remove the middle part of the prompt when it exceeds the `max_length` of the model

maxime-louis commented on config/dataset/frames_wiki2023.yaml:
```diff
@@ -0,0 +1,16 @@
+        init_args:
+            _target_: modules.dataset_processor.Wikipedia2023
+            split: "train"
+            data_path: "/nfs/data/calmar/amyrzakha/data/wiki2023/chunked_dataset.tsv"
```
hard-coded path here: please remove

maxime-louis replied:
hard-coded path here: please remove

maxime-louis commented on config/dataset/narrativeqa.yaml:
```diff
@@ -0,0 +1,15 @@
+dev:
+    doc: 
+        init_args:
+            _target_: modules.dataset_processor.NarrativeQA
```
NarrativeQA VS NarrativeQA_full: what's the difference ?

maxime-louis commented on modules/dataset_processor.py:
```diff
@@ -229,9 +229,41 @@ def process(self):
+        dataset = datasets.load_dataset(hf_name, num_proc=self.num_proc)[self.split]
+
+        dataset = dataset.rename_column("Prompt", "content")
+        dataset = dataset.map(lambda example: {"id": str(example["Unnamed: 0"])})
```
you can merge these two map calls into one by returning a dict with two entries

maxime-louis commented on modules/dataset_processor.py:
```diff
@@ -229,9 +229,41 @@ def process(self):
+
+class FramesOracle(Processor):
+    def __init__(self, *args, **kwargs):
+        dataset_name = 'frames_oracle'
```
Readme for that dataset (on hugging face) on hf is empty: redact a minimal one (at least give column names and an example of (shortened) sample)

maxime-louis commented on modules/dataset_processor.py:
```diff
@@ -251,6 +283,43 @@ def map_fn(example):
+        super().__init__(*args, **kwargs, dataset_name=self.dataset_name)
+    
+    def process(self):
+        dataset = datasets.load_dataset("csv", data_files=[self.path], delimiter="\t", column_names=["id", "text", "title"])[self.split]
```
this is based on a csv ?

maxime-louis commented on modules/dataset_processor.py:
```diff
@@ -251,6 +283,43 @@ def map_fn(example):
+        return dataset
+
+
+class Wikipedia2023Full(Processor):
```
What is the 'Full' version? Please add comments to describe that.
Is it useful to have these two versions ?

maxime-louis commented on modules/dataset_processor.py:
```diff
@@ -229,9 +229,41 @@ def process(self):
 # Document processors
 # ---------------------------------------- #
+
+class FramesOracle(Processor):
```
Is there a way to use the same class as above ? Aren't the oracle documents included in the list of documents from regular Frames dataset ?

aidarmyrzakhan commented:
- Removed the Wikipedia dataset processor due to duplication with existing classes: Wikipedia2023_full and Wiki_monolingual_100w.
- Merged the dataset processors for Frames and FramesOracle into a single processor.
- Combined the dataset processors for NarrativeQA and NarrativeQA_full into a single processor.
- Updated the configurations for the following datasets: frames, frames_oracle, frames_wiki2023, narrativeqa, and narrativeqa_full.

Commands line examples:
```
python bergen.py retriever="oracle_provenance" generator='vllm_llama-3.2-3b-instruct' +generator.init_args.max_doc_len=25000 +overwrite_datasets=True dataset='frames_oracle' #to run the oracle for frames

python bergen.py retriever="oracle_provenance" generator='vllm_llama-3.2-3b-instruct' +generator.init_args.max_doc_len=25000 +overwrite_datasets=True dataset='narrativeqa_full' #to run the full text oracle from NarrativeQA

python bergen.py retriever="oracle_provenance" generator='vllm_llama-3.2-3b-instruct' +overwrite_datasets=True dataset='narrativeqa' #to run the summary oracle from NarrativeQA

python bergen.py retriever="splade-v3" reranker="debertav3" generator='vllm_llama-3.2-3b-instruct' dataset='frames_wiki2023' #to run Frames with wikipedia and retriever and reranker

python bergen.py retriever="oracle_provenance" generator='vllm_llama-3.2-3b-instruct' +generator.init_args.use_middle_truncation=true +overwrite_datasets=True dataset='narrativeqa_full' #to use middle_truncation strategy to drop the middle part of the document to fit context length
```
